packages/pages/src/plugins/urlWriteback/yext.ts

```diff
@@ -0,0 +1,81 @@
+interface EntityProfile {
+  [field: string]: ProfileValue;
+
+  meta?: {
+    accountId: string;
+    uid: string;
+    id: string;
+    timestamp: string;
+    folderId: string;
+    language: string;
+    countryCode: string;
+    entityType: string;
+  };
+
+  name?: string;
+}
+
+type ProfileValue =
+  | string
+  | number
+  | boolean
+  | null
+  | undefined
+  | ProfileValue[]
+  | { [k: string]: ProfileValue };
+
+interface ApiResponse<T> {
+  meta: {
+    uuid: string;
+    errors: {
+      code: number;
+      message: string;
+      type: string;
+    }[];
+  };
+  response: T;
+}
+
+const API_BASE_PROD = "https://api.yext.com/v2/accounts/me/";
+const API_BASE_SBX = "https://api-sandbox.yext.com/v2/accounts/me/";
+
+export function buildApiUrl(base: string, path: string, params?: Record<string, string>) {
+  const result = new URL(path, base);
+  for (const k in params) {
+    result.searchParams.append(k, params[k]);
+  }
+  return result.toString();
+}
+
+// TODO: when the official typescript API client has more
+// functionality, switch to using that: https://github.com/yext/yext-api-ts
+export async function updateEntity<T extends EntityProfile>(
+  id: string,
+  locale: string,
+  body: EntityProfile,
+  apiKey: string,
+  options?: {
+    env?: "prod" | "sbx",
+    v?: string,
+  }
+): Promise<T> {
+  const URL_BASE = options?.env === "sbx"
+    ? API_BASE_SBX
+    : API_BASE_PROD;
+
+  const req = new Request(buildApiUrl(URL_BASE, `entityprofiles/${id}/${locale}`, {
+    api_key: apiKey,
+    v: options?.v || "20220903",
+  }), {
+    method: "PUT",
+    body: JSON.stringify(body),
+    headers: { "Content-Type": "application/json; charset=utf-8" },
+  });
+  const response = await fetch(req);
+  if (response.status < 200 || response.status >= 300) {
+    const responseBody = await response.json() as ApiResponse<T>;
+    throw responseBody.meta.errors[0].message
```

Comment:
Curious if this would be visible anywhere in the platform

Reply:
Worst case it's visible in the snowflake logs. Bryan has told me it's also visible in the platform on the deploys but I haven't seen exactly where.

Reply:
[Example](https://www.yext.com/s/1089287/yextsites/37118/branch/914/deploy/6gr6dp3pqu/stream/01GBWRH21XYPG2KDW3G65VRMQ2) - see the console log for plugin invocation section

Reply:
Do logs for all of the plugins (onUrlChange, OnPageGenerate, w/e) show up in that section?

Reply:
Any invocations associated with an activity should show up.

Reply:
Very cool that's where I was hoping it would go